control-plane/connect-inject/endpoints_controller.go

```diff
@@ -628,6 +636,7 @@ func (r *EndpointsController) createGatewayRegistrations(pod corev1.Pod, service
 		MetaKeyKubeServiceName: serviceEndpoints.Name,
 		MetaKeyKubeNS:          serviceEndpoints.Namespace,
 		MetaKeyManagedBy:       managedByValue,
+		MetaKeySyntheticNode:   "true",
```

Comment:
Would it be useful to set all the `MetaKeySyntheticNode` in this file and the test file using a const? In case we need to change the state in the future?